### PR TITLE
Task 38

### PR DESCRIPTION
**Pull request by ghost**
Additional task to contribute. 

The task is to find the shortest path in a directed weighted graph. 

There are only a few tests, as the task itself is quite wordy and complicated in implementation. 

The problem is with auxiliary class Path. It's created only for testing, so maybe it should be moved to directory with tests? Waiting for your advise.

### CODE REVIEW

File: src/main/java/com/epam/university/java/core/task038/Path.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.epam.university.java.core.task038;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import static java.lang.Math.pow;
+
+public class Path {
```
Comment (aabarmin) [inline src/main/java/com/epam/university/java/core/task038/Path.java]: Hi, 

It's better moving this class to the `src/main/test` and another piece of advice is to rename this class to something like `PathInfo`.

**Comment by aabarmin** [comment]
In common, this looks great. 

Thank you for contributing!

File: src/main/java/com/epam/university/java/core/task038/Path.java (hunk 1 above)
Comment (ghost) [reply src/main/java/com/epam/university/java/core/task038/Path.java]: Thank you for your advice. Requested work is done in a new commit.

**Comment by aabarmin** [comment]
Thanks, merged.

**Pull request merged**